### DISCUSSION TITLE
Customize Log4j 2 SLF4J Binding

### DISCUSSION THREAD

**Discussion by originalrusyn** [apache/logging-log4j2#1807] [Q&A]
Hi guys,

I would like to create custom Logger based on Log4j 2 SLF4J Binding to apply it globally instead of Log4j 2 SLF4J Binding in my app.

So, I've created decorators by implement

org.slf4j.Logger
org.slf4j.ILoggerFactory
org.slf4j.spi.SLF4JServiceProvider

interfaces for corresponded classes of https://github.com/apache/logging-log4j2/tree/2.x/log4j-slf4j2-impl

But I can't use my implementation because slf4j2 expects single SLF4JServiceProvider only https://github.com/qos-ch/slf4j/blob/7c164fab8d54f823dd55c01a5a839c153f578297/slf4j-api/src/main/java/org/slf4j/LoggerFactory.java#L399

So, is it possible to make Log4j 2 SLF4J Binding to be customizable?

**Comment by ppkarwasz** [comment]
@originalrusyn,

In SLF4J 2.0.9 Ceki introduced a `slf4j.provider` system property to give the FQCN of the provider. See [SLF4J-450](https://jira.qos.ch/browse/SLF4J-450) for details.
